Chatkit/Repositories/Change Reasons/JoinedRoomsRepositoryChangeReason.swift

```diff
@@ -0,0 +1,76 @@
+
+public extension JoinedRoomsRepository {
+    
+    /// Enumeration that describes changes introduced to the state of `JoinedRoomsRepository`.
+    enum ChangeReason {
+        
+        /// Notifies the receiver that the current user has been added to a room.
+        ///
+        /// - Parameters:
+        ///     - room: The room joined by the user.
+        case addedToRoom(room: Room)
+        
+        /// Notifies the receiver that the current user has been removed from a room.
+        ///
+        /// - Parameters:
+        ///     - room: The room which the user is no longer a member of.
+        case removedFromRoom(room: Room)
+        
+        /// Notifies the receiver that a room the current user is a member of has been updated.
+        ///
+        /// - Parameters:
+        ///     - updatedRoom: The new value of the room.
+        ///     - previousValue: The value of the room befrore it was updated.
+        case roomUpdated(updatedRoom: Room, previousValue: Room)
+        
+        /// Notifies the receiver that a read state summary of a room the current user is a member of has been updated.
+        ///
+        /// - Parameters:
+        ///     - updatedRoom: The new value of the room.
+        ///     - previousValue: The value of the room befrore it was updated.
+        case readStateUpdated(updatedRoom: Room, previousValue: Room)
```

Comment:
Regarding the variable labels here, was the naming a conscious choice?  I think I would expect `previousRoom` as opposed to `previousValue`.

Reply:
The name of this parameter has been copied the public API of the Android SDK. We should probably make this consistent on both platforms.

Reply:
As we just discussed with the team, please leave this as it is.